### PR TITLE
Fix 4.0-5689 regression (AR codes, patches)

### PR DESCRIPTION
**Pull request by JosJuice**
CreateCodeTab, ARCodeAddEdit and PatchAddEdit need to be able to modify arCodes/onFrame. I'm not familiar with this code, so this definitely needs reviewing.

### CODE REVIEW

File: Source/Core/DolphinWX/ARCodeAddEdit.cpp [hunk 1]
```diff
@@ -44,8 +44,8 @@ CARCodeAddEdit::CARCodeAddEdit(int _selection, const std::vector<ActionReplay::A
 	}
 	else
 	{
-		currentName = StrToWxStr(arCodes.at(selection).name);
-		tempEntries = arCodes.at(selection);
+		currentName = StrToWxStr(arCodes->at(selection).name);
+		tempEntries = arCodes->at(selection);
```
Comment (lioncash) [inline Source/Core/DolphinWX/ARCodeAddEdit.cpp]: These .at calls can just be turned into array subscripts. E.g.

```cpp
currentName = StrToWxStr(arCodes[selection].name);
tempEntries = arCodes[selection];
```

Comment (JosJuice) [inline Source/Core/DolphinWX/ARCodeAddEdit.cpp]: That doesn't seem to work. I get errors like these:

    error C2039: 'name' : is not a member of 'std::vector<ActionReplay::ARCode,std::allocator<_Ty>>'

Comment (magumagu) [inline Source/Core/DolphinWX/ARCodeAddEdit.cpp]: It would need to be `(*arcodes)[selection]`.  Not sure if that's an improvement.

Comment (heinermann) [inline Source/Core/DolphinWX/ARCodeAddEdit.cpp]: Why not use a reference instead of a pointer?

File: Source/Core/DolphinWX/ARCodeAddEdit.cpp [hunk 2]
```diff
@@ -57,8 +57,8 @@ CARCodeAddEdit::CARCodeAddEdit(int _selection, const std::vector<ActionReplay::A
 	EditCheatName->SetValue(currentName);
 
 	EntrySelection = new wxSpinButton(this);
-	EntrySelection->SetRange(1, ((int)arCodes.size()) > 0 ? (int)arCodes.size() : 1);
-	EntrySelection->SetValue((int)(arCodes.size() - selection));
+	EntrySelection->SetRange(1, ((int)arCodes->size()) > 0 ? (int)arCodes->size() : 1);
```
Comment (heinermann) [inline Source/Core/DolphinWX/ARCodeAddEdit.cpp]: Would std::max(arCodes->size(), 1); be more appropriate here?

File: Source/Core/DolphinWX/ARCodeAddEdit.cpp (hunk 1 above)
Comment (JosJuice) [inline Source/Core/DolphinWX/ARCodeAddEdit.cpp]: Per the style guide, pointers should be used instead of references if the data is modified.

**Comment by skidau** [comment]
@dolphin-emu-bot allowmerge

**Pull request merged**